backend/src/relationships/friendships/friendships.service.ts

```diff
@@ -48,8 +51,15 @@ export class FriendshipsService {
     return await this.friendshipRepository.save(friendship);
   }
 
-  findAll(query?: QueryFriendshipDto) {
-    return this.friendshipRepository.find({ where: query });
+  async findAll(
```

Comment:
If you had problems with this route, maybe it would be a good idea to change the block service now because it is implemented the same way, and we might forget otherwise.

Reply:
This route works, except for the `QueryFriendshipDto` which had typos. I corrected the same typos in `QueryBlockDto`. The rest of the changes are just to make it a paginated route, which would enable us to put the friends in an list component which make requests incrementally (let us imagine someone has a lot of friends). But a similar component for the blocks does not seem necessary, so maybe we do not need to change it.